### PR TITLE
add readme for prior

### PR DESCRIPTION
**Pull request by nousr**
@rom1504 I started this, feel free to give feedback!

✅ Drafted
- [x]     What is the prior and what can it do
- [x]     how to use a pretrained model
- [x]     How to prepare a dataset for it
- [x]     How to run a training
- [x]     What are good metrics

❓ Maybe this can remain in the main-readme or a decoder's readme? since the prior plugs into the decoder:
-  ~how to plug into the decoder~

### CODE REVIEW

**Comment by nousr** [comment]
> How to prepare a dataset for it

Is there good EmbeddingReader documentation we can point to for this?

File: prior.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+## Training
+
+### Overview
+
+Training the prior is a relatively straightforward process thanks to the Trainer base class. The major step that is required of you is preparing a dataset in the format that EmbeddingReader expects. Having pre-computed embeddings massively increases training efficiency and is generally recommended as you will likely benefit from having them on hand for other tasks as well. Once you have a dataset, you are ready to move onto configuration
+
+## Dataset
+
+To train the prior, it is highly recommended to use precomputed embeddings for the images. To obtain these for a custom dataset, you can leverage [img2datset](https://github.com/rom1504/img2dataset) to pull images from a list of URLs and [clip_retrieval](https://github.com/rom1504/clip-retrieval#clip-inference) for genertating the actual embeddings that can be used in the prior's dataloader.
```
Comment (rom1504) [inline prior.md:142]: img2datset genertating

File: prior.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+## Configuration
+
+The configuration file allows for you to easily track and reproduce experiments. It is a simple JSON file that will specify the architecture, dataset, and training parameters. For more information and specifics please see the configuration README.
```
Comment (rom1504) [inline prior.md:146]: would be nice to add a link to the relevant section

**Comment by rom1504** [comment]
nice!

File: prior.md (hunk 1 above)
Comment (nousr) [reply prior.md]: https://github.com/lucidrains/DALLE2-pytorch/pull/159/commits/9bdb92cf20d211e6bb20bc5b0fbc56d8685913cc

File: prior.md (hunk 2 above)
Comment (nousr) [reply prior.md:146]: will do!

**Pull request merged**